### DISCUSSION TITLE
How to mapping GPIO to be PINT for wakeup source in LPC55s69

### DISCUSSION THREAD

**Discussion by MarsMSWu** [zephyrproject-rtos/zephyr#76574] [Q&A] [answered]
Hi all,

I'm try to using GPIO0_6 under PINT group for wakeup source.

Study the code from MCUXpresso IDE, it's GINT example and shows some steps are necessary
1. Initialize GINT
2. Set GINT control, likes trigger level or callback function
3. Connect which GPIO group to connect to GINT
4. Enter sleep and set parameter_3 is WAKEUP_GPIO_GLOBALINT0 to be wakeup source

I follow the GINT flow and try to modify a version for PINT, because I only want to use PIO0_6 to be wakeup source, so my steps as below
1. Initialize PINT, PINT just point to 0x5000 4000 address
2. In my understanding, this step should connect PIO0_6 to PINT (not find a API likes GINT_SetCtrl in SDK)
3. Set configure for PINT, likes rising edge trigger without callback

Is anyone can shsre more information to me ?

Thanks, Mars

**Comment by jacob-wienecke-nxp** [comment, marked as answer]
Hi, Mars

I think you will find the Zephyr Button sample under "zephyr/samples/basic/button" useful. The button example configures the PINT for interrupt. Within the context of LPC55S69 Sleep Mode, any interrupt will wake the device from sleep.

Thanks, Jacob

**Reply by MarsMSWu** [reply]
Hi Jacob,

I see the sample...it seems like just define button with interrupt function
Do you mean "gpio_pin_interrupt_configure_dt(&button, GPIO_INT_EDGE_TO_ACTIVE);" can wakeup system from sleep or deep-sleep ?
![image](https://github.com/user-attachments/assets/c98adf77-dc87-4d6e-a2c7-60c4e39821df)

If I use "POWER_EnterDeepSleep( )" to configure wakeup source, how to map the &button to correct WAKEUP source group ?
![image](https://github.com/user-attachments/assets/a7c36a8d-f267-4032-9679-b705a137f4cd)

Or...I don't need use "POWER_EnterDeepSleep( )" to configure wakeup source ?

Thanks, Mars

**Reply by jacob-wienecke-nxp** [reply]
Mars,

Deep Sleep is a configurable mode within LPC55s69, but it is not supported for this device in zephyr. The zephyr-supported power mode for LPC55s69 is sleep. Therefore, the function "POWER_EnterDeepSleep()" should not be required.

The table (323) from the RM shows configurable wakeup sources from Deep-sleep, Power-down, and Deep power-down modes. For sleep mode, no wake source configuration is required. From RM section 13.3.3.3 _"A CPU sleep-mode is exited automatically when an interrupt enabled by the NVIC arrives at the processor or a reset occurs."_ This means that, as long as the interrupt is enabled via the NVIC, the interrupt will wake from sleep.

Therefore, configuring a pin interrupt (which will create an interrupt in the NVIC) is sufficient to wake the part from sleep when pin interrupt occurs.

Thanks, Jacob

**Reply by MarsMSWu** [reply]
Hi Jacob,

It very thanks for your feedback...it looks like sleep mode is Zephyr feature and auto wakeup by interrupt event.
Other mode is chip feature, if I want more lower power consumption I should enter to deep-sleep...

By this question title, I think the button example should be correct for me
I will keep try how to make chip enter deep-sleep and wakeup by IO configure...

Thanks, Mars
